app/addons/documents/mango/mango.actions.js

```diff
@@ -41,18 +55,54 @@ function (app, FauxtonAPI, Documents, ActionTypes, Stores, IndexResultsActions)
       });
 
       mangoIndex.save().then(function (res) {
-        var msg = res.result === 'created' ? 'Index created' : 'Index already exits',
-            url = FauxtonAPI.urls('mango', 'index-app', options.database.safeID());
+        var msg = res.result === 'created' ? 'Index created.' : 'Index already exits.',
+            url = FauxtonAPI.urls('mango', 'query-app', options.database.safeID());
 
         FauxtonAPI.addNotification({
-          msg:  msg,
+          msg:  msg + ' Redirect to search...',
           type: 'success',
           clear: true
         });
 
-        IndexResultsActions.reloadResultsList();
+        FauxtonAPI.dispatch({
+          type: ActionTypes.MANGO_NEW_QUERY_FIND_CODE_FROM_FIELDS,
+          options: {
+            fields: queryCode.index.fields
+          }
+        });
+
+        window.setTimeout(function () {
+          FauxtonAPI.navigate(url);
+          FauxtonAPI.addNotification({
+            msg:  'Feel free to search now.',
```

Comment:
Maybe we should say something like `Index is ready for querying` or `Index is ready` or something like that. I'm not sure `Feel free to search now.` is the correct tone.